src/components/AlphabetFilter/AlphabetFilter.tsx

```diff
@@ -0,0 +1,197 @@
+        refCurrentLetter.current = currentLetter;
+      }
+    }, [currentLetter, onClick]);
+
+    /**
+     * This function is only called when clicking on a link should
+     * not update the URL or refresh the page.
+     */
+    const handleOnClick = (
+      e: React.MouseEvent | React.KeyboardEvent,
+      clickedLetter: string
+    ) => {
+      e.preventDefault && e.preventDefault();
+      if (selectedLetter === clickedLetter) return;
+      setSelectedLetter(clickedLetter);
+      onClick && onClick(clickedLetter);
+    };
+
+    const getButtonElement = (item) => {
+      const isSelectedLetter =
+        selectedLetter === item.value && item.value !== "showAll";
+      const isShowAll = item.value === "showAll";
+      const isButtonDisabled =
+        isDisabled ||
+        (activeLetters && !activeLetters.includes(item.value) && !isShowAll);
+
+      const buttonStyles = isSelectedLetter
+        ? { ...styles, border: "1px solid gray" }
+        : isShowAll
+        ? {
+            ...styles,
+            fontWeight: "normal !important",
+            whiteSpace: "nowrap",
+            marginLeft: "var(--nypl-space-m)",
+          }
+        : styles;
+
+      return (
+        <Button
+          aria-current={isSelectedLetter ? "page" : null}
+          aria-label={
+            item.text === "Show All" ? item.text : "Page " + item.text
+          }
+          buttonType="link"
+          id={`filter-${item.value}`}
+          isDisabled={isButtonDisabled}
+          __css={buttonStyles}
+          onClick={(e: React.MouseEvent | React.KeyboardEvent) => {
+            handleOnClick(e, item.value);
+          }}
+        >
+          {item.text}
+        </Button>
+      );
+    };
+
+    const getFilterLetters = () => {
+      if (isLargerThanMobile) {
+        return filterButtons.map((item) => {
+          return getButtonElement(item);
+        });
+      }
+      // split letters into rows of 10 + Show All
+      const mobileRows = filterButtons.reduce((rows, key, index) => {
+        if (index !== filterButtons.length - 1) {
+          return (
+            (index % 10 === 0
+              ? rows.push([key])
+              : rows[rows.length - 1].push(key)) && rows
+          );
+        } else {
+          return rows.push([key]) && rows;
+        }
+      }, []);
+      return (
+        <VStack sx={styles.mobileStack}>
+          {mobileRows.map((row, index) => {
+            return (
+              <Box sx={styles.filterRow} key={index}>
+                {row.map((item) => {
+                  return getButtonElement(item);
+                })}
+              </Box>
+            );
+          })}
+        </VStack>
+      );
+    };
+
+    return (
+      <Box
+        as="nav"
+        role="navigation"
+        id={id}
+        className={className}
+        ref={ref}
+        __css={styles}
+        {...rest}
+      >
+        {headingText && <Heading size="tertiary">{headingText}</Heading>}
```

Comment:
Sorry, there was some confusion here due to the Figma.  This should be `size=“secondary”` or the `size` prop should be removed altogether.  Using `ComponentWrapper` will take care of this.